org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/compiler/OnTheFlyJavaCompiler2.java

```diff
@@ -0,0 +1,187 @@
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("Code was:");
+				richString.newLine();
+				richString.append(code);
+				richString.newLineIfNotEmpty();
+				throw new IllegalArgumentException(richString.toString());
+			}
+			return result.getClassLoader().loadClass(classname);
+		} catch (Throwable t) {
+			if (t instanceof ClassNotFoundException) {
+				ClassNotFoundException e = (ClassNotFoundException) t;
+				StringConcatenation richString = new StringConcatenation();
+				richString.append("Couldn\'t load \'");
+				richString.append(classname);
+				richString.append("\' ");
+				richString.newLineIfNotEmpty();
+				richString.append("source :");
+				richString.newLine();
+				richString.append("\t");
+				richString.append(code, "\t");
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("PROBLEMS : ");
+				richString.newLine();
+				richString.append("\t");
+				richString.append(Joiner.on("\n").join(result.getCompilationProblems()), "\t");
+				richString.newLineIfNotEmpty();
+				throw new IllegalStateException(richString.toString(), e);
+			} else {
+				throw Exceptions.sneakyThrow(t);
+			}
+		}
+	}
+
+	private String toJavaFile(String string) {
+		return string.replace(".", "/") + ".java";
+	}
+
+	public Map<String, Class<?>> compileToClasses(Map<String, String> sources) {
+		InMemoryJavaCompiler.Result result = inMemoryCompiler
+				.compile(((JavaSource[]) Conversions.unwrapArray(
+						Iterables.transform(sources.entrySet(),
+								(Map.Entry<String, String> it) -> new JavaSource(toJavaFile(it.getKey()), it.getValue())),
+						JavaSource.class)));
+		try {
+			if (IterableExtensions.exists(result.getCompilationProblems(), CategorizedProblem::isError)) {
+				StringConcatenation richString = new StringConcatenation();
+				richString.append("Java code compiled with errors:");
+				richString.newLine();
+				richString.append(
+						Joiner.on("\n").join(IterableExtensions.filter(result.getCompilationProblems(), CategorizedProblem::isError)));
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("Code was:");
+				richString.newLine();
+				richString.append("=========");
+				richString.newLine();
+				richString.append(Joiner.on("\n=========\n").join(sources.values()));
+				richString.newLineIfNotEmpty();
+				richString.append("=========");
+				richString.newLine();
+				throw new IllegalArgumentException(richString.toString());
+			}
+			ClassLoader classLoader = result.getClassLoader();
+			return IterableExtensions.toMap(Iterables.transform(sources.keySet(), (String it) -> {
+				try {
+					return classLoader.loadClass(it);
+				} catch (ClassNotFoundException e) {
+					throw Exceptions.sneakyThrow(e);
+				}
+			}), Class::getName);
+		} catch (Throwable t) {
+			if (t instanceof ClassNotFoundException) {
+				ClassNotFoundException e = (ClassNotFoundException) t;
```

Comment:
Use ordinary string concatnation.

Reply:
did not want to change any semantics on a deprecated class